Add theming to widgets: Dynamic color (Material You) and Transparent

jpelgrom opened the pull request:
<!-- Thank you for submitting a Pull Request and helping to improve Home Assistant. Please complete the following sections to help the processing and review of your changes. Please do not delete anything from this template. -->

## Summary
<!-- Provide a brief summary of the changes you have made and most importantly what they aim to achieve -->
Adds theming options to most widgets:
 - **Dynamic color**: uses colors from the device theme (eg. Material You) on supported Android 12+ devices. Default for new widgets on devices that support it.
 - **Light/dark theme**: the existing widget theme. Default for new widgets on devices that don't support dynamic color.
 - **Transparent**: removes the widget background, user can choose whether the content should be black or white.

The text size for template widget can now also be set.

More minor changes for Android 12 to improve the widget theme:

 - Album art now is correctly clipped by the widget corner radius
 - `android:previewLayout` is set to the dynamic color layout file, to provide dynamic previews and better match the widget that'll be added
 - Suggested size for entity state and template widgets is updated to 2x1, to match the preview images
 - When using dynamic color, the system widget corner radius is used

Resolves #1852: this PR adds options for background and font size, and additional formatting can be done using HTML tags.

Any widgets that are already added won't change with this PR, they are migrated to settings that match the existing widget theme. The camera widget doesn't get any theming options because you should only see a camera image most of the time. The media player widget doesn't get a transparent theme because floating text and buttons next to album art doesn't make sense.

## Screenshots
<!-- If this is a user-facing change not in the frontend, please include screenshots in light and dark mode. -->
Theming options when creating a widget. Also shows the template widget text size option.
|Light|Dark|
|-----|-----|
|![image](https://user-images.githubusercontent.com/8148535/166332430-bb6d4381-20a5-4143-ac63-3160621ffbb8.png)|![image](https://user-images.githubusercontent.com/8148535/166332461-44bbb3f0-8799-4721-b381-e4e0046f548a.png)|

When the 'Transparent' theme is selected, an additional option for the widget's text/icon color will be shown.
|Light|Dark|
|-----|-----|
|![image](https://user-images.githubusercontent.com/8148535/166332604-162fa7a5-f730-48ee-9a06-3422d0f28237.png)|![image](https://user-images.githubusercontent.com/8148535/166332619-b5fea19b-7677-4b37-a455-ab56c0d50a77.png)|

Transparent widgets, from left to right: entity state, button, template
|'White' text color|'Black' text color|
|-----|-----|
|![image](https://user-images.githubusercontent.com/8148535/166152331-5f4a7281-be1b-4aa4-a665-4aa348a9835d.png)|![image](https://user-images.githubusercontent.com/8148535/166152335-d839b968-a89d-47b2-b124-d49f2fc665d5.png)|

Dynamic color widgets, a few examples. The colors used match widgets provided by Google apps. The corner radius for these widgets also matches the system value, which for this device is larger than the value normally used by the app. (Color varies slightly between widgets because it actually uses the widget location on the background!)
|Light|![image](https://user-images.githubusercontent.com/8148535/166152777-467b4ac9-7981-421e-bf5f-a11b17d1f284.png)|![image](https://user-images.githubusercontent.com/8148535/166152780-06cd31a0-6837-4c64-88b6-418a893a1280.png)|![image](https://user-images.githubusercontent.com/8148535/166152781-ab7d5626-f6c4-4056-89ea-6f0cdb6bd0a7.png)|![image](https://user-images.githubusercontent.com/8148535/166152789-45e8c3f1-ea99-4167-932f-6c669ee70b31.png)|
|-----|-----|-----|-----|-----|
|Dark|![image](https://user-images.githubusercontent.com/8148535/166152796-96f8f7e4-cc35-4fd2-a53f-8c3fd8c2b726.png)|![image](https://user-images.githubusercontent.com/8148535/166152798-fc537845-a228-4702-97af-e955d4e09d04.png)|![image](https://user-images.githubusercontent.com/8148535/166152801-88f88a3a-91b7-4005-9acd-383551312a19.png)|![image](https://user-images.githubusercontent.com/8148535/166152806-fb45a866-4d82-48cd-8cd1-5f07f7746705.png)|

Launcher providing dynamic previews of widgets:
|Light|Dark|
|-----|-----|
|![image](https://user-images.githubusercontent.com/8148535/166152942-46630708-00e3-42e4-800e-64fe467f751b.png)|![image](https://user-images.githubusercontent.com/8148535/166152945-e55f7173-94f4-4462-9d2c-a81d86493f9c.png)|

## Link to pull request in Documentation repository
<!-- Pull requests that add, change or remove functionality must have a corresponding pull request in the Companion App Documentation repository (https://github.com/home-assistant/companion.home-assistant). Please add the number of this pull request after the "#" -->
Documentation: home-assistant/companion.home-assistant#740

## Any other notes
<!-- If there is any other information of note, like if this Pull Request is part of a bigger change, please include it here. -->

JBassett reviewed (changes requested):
My only issue is the radio button alignment, it shouldn't just be right aligned.

jpelgrom commented:
> My only issue is the radio button alignment, it shouldn't just be right aligned.

Updated alignment for the text/icon color radio buttons to left aligned (for ltr locales), and switched to a dropdown menu for the theme selection because it didn't look right to me when changing the alignment.
